### PR TITLE
Handle unresolvable cmdcomd names from the database

### PR DESCRIPTION
**Pull request by daveboutcher**

### CODE REVIEW

File: src/main/java/com/ocient/jdbc/XGConnection.java [hunk 1]
```diff
@@ -944,10 +945,14 @@ private void saveSecondaryInterfaces(final List<String> newCmdComps, final List<
...
-					connInfo = addr.toString().substring(addr.toString().indexOf('/') + 1) + ":" + connPort;
-					this.secondaryInterfaces.get(i).add(connInfo);
+				try {
+					final InetAddress[] addrs = InetAddress.getAllByName(connHost);
+					for (final InetAddress addr : addrs) {
+						connInfo = addr.toString().substring(addr.toString().indexOf('/') + 1) + ":" + connPort;
+						this.secondaryInterfaces.get(i).add(connInfo);
+					}
+				} catch (final UnknownHostException e) {
+					LOGGER.log(Level.DEBUG, "Cannot resolve hostname %s received from database", connHost);
```
Comment (jwilliams-ocient) [inline src/main/java/com/ocient/jdbc/XGConnection.java:955]: LGTM, I see that we swallow this exception in pyocient as well.

One question: do we need an array bounds check when the now potentially-empty list is consumed? I can't tell for certain, but it looks like this code assumes all of the list elements in `XGConnection#secondaryInterfaces` are of equal length. 
https://github.com/Xeograph/ocient_jdbc/blob/78ca3419144aafa1ffd557de7d082beb07de4f5e/src/main/java/com/ocient/jdbc/XGConnection.java#L3316-L3322

Comment (daveboutcher) [reply src/main/java/com/ocient/jdbc/XGConnection.java]: this whole thing is fragile, but if names for one mode (SQL0, say) are resolvable, but names for another  node (SQL!) are not, this is just going to be screwed in any case

**Comment by daveboutcher** [comment]
before fix:
```
Ocient> connect to jdbc:ocient://10.10.112.42:4050/system;logfile=/tmp/jdbc.log;loglevel=DEBUG;
Failed to connect to 10.10.112.42:4050(bigdog.fooxxxxx.com: Name or service not known)
Error: bigdog.fooxxxxx.com: Name or service not known
Ocient> 
```

after fix:
```
Ocient> connect to jdbc:ocient://10.10.112.42:4050/system;logfile=/tmp/jdbc.log;loglevel=DEBUG;
Connected to jdbc:ocient://10.10.112.42:4050/system
Ocient> 
```

and the log has 
```
2022-10-21T05:40:31.000228 [1] com.ocient.jdbc.XGConnection saveSecondaryInterfaces INFO: Clearing and adding new secondary interfaces 
2022-10-21T05:40:31.000230 [1] com.ocient.jdbc.XGConnection saveSecondaryInterfaces INFO: Cannot resolve hostname bigdog.fooxxxxx.com received from database 
2022-10-21T05:40:31.000232 [1] com.ocient.jdbc.XGConnection saveSecondaryInterfaces INFO: Cannot resolve hostname smalldog.fooxxxxxx.com received from database 
2022-10-21T05:40:31.000232 [1] com.ocient.jdbc.XGConnection saveSecondaryInterfaces INFO: Cannot resolve hostname smalldog.fooxxxxxx.com received from database 
```

**Review by jwilliams-ocient — APPROVED** [review]
Ship it!

**Pull request merged**